crates/primitives/starknet/src/tests/transaction.rs

```diff
@@ -373,3 +376,72 @@ fn test_event_wrapper_builder_with_event_content() {
 
     pretty_assertions::assert_eq!(event_wrapper, expected_event);
 }
+
+#[test]
+fn test_try_into_deploy_account_transaction() {
+    fn helper(
+        array_size: usize,
+        calldata_size: usize,
+    ) -> Result<(DeployAccountTransaction, DeployAccountTransaction), BroadcastedTransactionConversionErrorWrapper>
+    {
+        let signature: Vec<FieldElement> = vec![FieldElement::default(); array_size];
+        let constructor_calldata: Vec<FieldElement> = vec![FieldElement::default(); calldata_size];
+
+        let input = BroadcastedDeployAccountTransaction {
+            signature,
+            constructor_calldata,
+            // `FieldElement` can be trivially converted to `Felt252Wrapper` so no need to test them
+            max_fee: FieldElement::default(),
+            nonce: FieldElement::default(),
+            contract_address_salt: FieldElement::default(),
+            class_hash: FieldElement::default(),
+        };
+
+        let output: DeployAccountTransaction = input.try_into()?;
+
+        let expected_signature = bounded_vec![Felt252Wrapper::default(); array_size];
+        let expected_calldata = bounded_vec![Felt252Wrapper::default(); calldata_size];
+
+        let expect_output = DeployAccountTransaction {
+            version: 1_u8,
+            calldata: expected_calldata,
+            signature: expected_signature,
+            nonce: Felt252Wrapper::from(FieldElement::default()),
+            salt: Felt252Wrapper::from(FieldElement::default()),
+            account_class_hash: Felt252Wrapper::from(FieldElement::default()),
+            max_fee: Felt252Wrapper::from(FieldElement::default()),
+        };
+
+        Ok((output, expect_output))
+    }
+
+    let zero_len = helper(0, 0).expect("shouldn't fail");
+    pretty_assertions::assert_eq!(zero_len.0, zero_len.1);
+
+    let one_len = helper(1, 1).expect("shouldn't fail");
+    pretty_assertions::assert_eq!(one_len.0, one_len.1);
+
+    // Cannot run this tests because currently `MaxArraySize` and `MaxCalldataSize` is very large
+    // so its not possible to make allocation for such large `Vec` and `BoundedVec`
```

Comment:
You can mock the max values to do these tests.

Reply:
can you explain how can I do that, because this max values are set inside [`DeployAccountTransaction`](https://github.com/keep-starknet-strange/madara/blob/f887287a1ae8b3fe1be00ef39358bab3040f42f0/crates/primitives/starknet/src/transaction/types.rs#L352-L367) and its not generic over that value.

Reply:
You can find the test I mentioned on TG [here](https://github.com/keep-starknet-strange/madara/blob/main/crates/primitives/starknet/src/tests/transaction.rs#L269).
Indeed here it works because `MaxArraySize` is 10001 and wouldn't work with `u32::MAX`. So we need to find some workaround to mock these `Max` values within tests.

Reply:
will use [this](https://doc.rust-lang.org/reference/conditional-compilation.html#test) to change max values while running tests